cpp/dolfinx/fem/assemble_vector_impl.h

```diff
@@ -26,6 +26,56 @@
 namespace dolfinx::fem::impl
 {
 
+/// Extract a sub-set of entities whose cell index (local to process) is marked
+///
+/// @param[in] entities The entities.
+/// @param[in] cell_indicator An indicator vector. The size of cell_indicator
+/// should be smaller than max(entities). A cell is marked if it contains a
+/// non-zero value.
+/// @tparam stride  The number of entries in each entity (1 for cells, 2 for
+/// exterior facets, 4 for interior facets)
+/// @returns A list of sub-entities
+template <std::size_t stride>
+std::vector<std::int32_t>
+_extract_entities(std::span<const std::int32_t> entities,
+                  std::span<const std::int8_t> cell_indicator)
+{
+  std::vector<std::int32_t> sub_entities;
+  sub_entities.reserve(entities.size());
+  if (stride == 1)
```

Comment:
Should this be `constexpr`?

Is there a measurable performance benefit to templating over `stride`?

Reply:
This was a const-expr: https://github.com/FEniCS/dolfinx/pull/2395#discussion_r1023622907, but you asked me to remove this.

Reply:
Yes, but the template parameter is still there. Can it be done simply without a template parameter and without special cases?